src/domain/tagset/tagset.service.ts

```diff
@@ -113,4 +113,11 @@ export class TagsetService {
     tagsetable.tagsets?.push(newTagset as ITagset);
     return newTagset;
   }
+
+  hasTag(tagset: ITagset, tagToCheck: string): boolean {
+    for (const tag of tagset.tags) {
+      if (tag === tagToCheck) return true;
```

Comment:
tagset.tags is normal array. Use [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) instead of iterating over the array.

Reply:
Comparing string to ITagset, ignore my comments. It's ok.